packages/trie/src/trie/trie.ts

```diff
@@ -749,7 +759,7 @@ export class Trie {
       root: this.root(),
     })
     if (includeCheckpoints && this.hasCheckpoints()) {
-      trie._db.checkpoints = [...this._db.checkpoints]
+      trie._db.setCheckpoints(this._db.checkpoints)
```

Comment:
Ah right. Yes this is nice. I think you should probably target the branch of #2203 here?

Reply:
Yes, it does not really matter anyways.